core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/bundle/BundleRestService.java

```diff
@@ -0,0 +1,217 @@
+@RestController
+@RequestMapping("/bundles")
+public class BundleRestService extends AbstractRestService {
+	
+	public BundleRestService() {
+		super(ResourceDocument.Fields.SORT_FIELDS);
+	}
+	
+	@ApiOperation(
+			value="Retrieve bundles", 
+			notes="Returns a collection resource containing all/filtered registered bundles."
+				+ "<p>Results are by default sorted by ID."
+				+ "<p>The following properties can be expanded:"
+				+ "<p>"
+				+ "&bull; resources() &ndash; this list of resources this bundle contains"
+		)
+		@ApiResponses({
+			@ApiResponse(code = 200, message = "OK", response = CodeSystems.class),
+			@ApiResponse(code = 400, message = "Bad Request", response = RestApiError.class)
+		})
+		@GetMapping(produces = { AbstractRestService.JSON_MEDIA_TYPE })
+		public Promise<Bundles> searchByGet(final BundleRestSearch params) {
+			return BundleRequests.prepareSearchBundle()
+					.filterByIds(params.getId())
+					.filterByTerm(params.getTitle())
+					.setLimit(params.getLimit())
+					.setExpand(params.getExpand())
+					.setSearchAfter(params.getSearchAfter())
+					.sortBy(extractSortFields(params.getSort()))
+					.buildAsync()
+					.execute(getBus());
+		}
+	
+	@ApiOperation(
+			value="Retrieve bundles", 
+			notes="Returns a collection resource containing all/filtered registered bundles."
+				+ "<p>Results are by default sorted by ID."
+				+ "<p>The following properties can be expanded:"
+				+ "<p>"
+				+ "&bull; resources() &ndash; this list of resources this bundle contains"
+		)
+		@ApiResponses({
+			@ApiResponse(code = 200, message = "OK", response = CodeSystems.class),
+			@ApiResponse(code = 400, message = "Invalid search config", response = RestApiError.class),
+			@ApiResponse(code = 404, message = "Branch not found", response = RestApiError.class)
+		})
+		@PostMapping(value="/search", produces = { AbstractRestService.JSON_MEDIA_TYPE })
+		public Promise<Bundles> searchByPost(final BundleRestSearch params) {
+			return searchByGet(params);
+		}
+	
+	@ApiOperation(
+			value="Retrieve budnle by its unique identifier",
+			notes="Returns generic information about a single bundle associated to the given unique identifier.")
+	@ApiResponses({
+		@ApiResponse(code = 200, message = "OK"),
+		@ApiResponse(code = 404, message = "Not found", response = RestApiError.class)
+	})
+	@GetMapping(value = "/{bundleId}", produces = { AbstractRestService.JSON_MEDIA_TYPE })
+	public Promise<Bundle> get(
+			@ApiParam(value="The bundle identifier")
+			@PathVariable(value="bundleId") final String bundleId) {
+		return BundleRequests.prepareGetBundle(bundleId)
+				.buildAsync()
+				.execute(getBus());
+	}
+	
+	@ApiOperation(
+			value="Create a bundle",
+			notes="Create a new bundle with the given parameters"
+		)
+		@ApiResponses({
+			@ApiResponse(code = 201, message = "Created", response = Void.class),
+			@ApiResponse(code = 400, message = "Bundle already exists in the system", response = RestApiError.class)
```

Comment:
Already exists results in `409` error, not `400`, please update.
Also, add the missing `400` ApiResponse, which is for Bad Requests.